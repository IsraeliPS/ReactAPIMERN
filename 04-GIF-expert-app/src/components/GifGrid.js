// import React, { useEffect, useState } from 'react'
// import { getGifs } from '../helpers/getGifs'

import GifGridItem from '../components/GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

import PropTypes from 'prop-types'

function GifGrid ({ category }) {
  // const [images, setImages] = useState([])

  const { data: images, loading } = useFetchGifs(category)
  // useEffect(() => {
  //   getGifs(category)
  //     .then(setImages)
  // }, [category])

  // console.log(category)

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading</p>}

      <div className='card-grid'>
        {images?.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  )
}
// Hp9LfGIZYzHTfwoE4XLnq9H44dAGwi2t
export default GifGrid

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
