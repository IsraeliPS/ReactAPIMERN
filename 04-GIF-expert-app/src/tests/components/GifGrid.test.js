import React from 'react'
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import '../../setupTests'

import GifGrid from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas de <GifGrid/>', () => {
  const category = 'Fairy Tail'
  //   const wrapper = shallow(<GifGrid category={category} />)

  test('Debe mostrar el componente correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe mostrar items cuando se carga el useFetchGifs', () => {
    const gifs = [{
      id: 'abc',
      url: 'https://example/cosa.jpg',
      title: 'Este es un titulo'
    },
    {
      id: '123',
      url: 'https://example/cosa.jpg',
      title: 'Este es un titulo'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })
    const wrapper = shallow(<GifGrid category={category} />)
    // expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })
})
