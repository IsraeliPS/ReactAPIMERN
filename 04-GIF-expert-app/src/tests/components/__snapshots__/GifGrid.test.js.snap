// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGrid/> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
    Fairy Tail
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
